mono/monodevelop
# [Ide] Adds feature in MonoDevelop to pin opened tabs

## Feature

* Enables our document bar to **pin tabs**.
* Adds some menu items in the contextual menu like: "Close All Except Pinned", "Un/Pin Tab"
* Parity with VS windows

| <img height="80" src="https://user-images.githubusercontent.com/1587480/57515485-0f503b80-7313-11e9-8366-5e64451eec77.jpg"/>| <img height="80" src="https://user-images.githubusercontent.com/1587480/57516423-fe082e80-7314-11e9-9ec2-4a86752636be.png"/> |<img height="80" src="https://user-images.githubusercontent.com/1587480/57516669-7111a500-7315-11e9-9de4-06c2da816923.png"/> |


![dragdrop](https://user-images.githubusercontent.com/1587480/57498598-b2d22980-72dc-11e9-93b9-defbc78b3f42.gif)

This feature allows to lock some of your most used windows (tabs)
to easily detect and open like VS2013 added.

Fixes VSTS #890509 - Add support for pinned tabs
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/890509

Fixes VSTS #945511 - Use the same pin from Start window at a smaller size for Tab pinning experience

## How to test

1) Feature is enabled by default
2) When enabled a pin icon will appear in every tab on over
3) Try pin/unpin tabs and ensure behaviour is correct
the reordering only works over same kind of tabs groups
4) Try the new commands created:
- Close All Except pinned
- Pin Tab
5) Test commands modified
- Close All
6) Ensure commands don't work with feature disabled


Fixes VSTS #890509 - Add support for pinned tabs
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/890509
Fixes VSTS #890508 - Add support for pinned tabs
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/890508
Includes fixes from:
#2487 and #2449 and #2606

## Mac
![image](https://user-images.githubusercontent.com/1587480/57499266-61776980-72df-11e9-9ec3-ce09a9120bec.png)

## Windows
![image](https://user-images.githubusercontent.com/1587480/57499226-2ecd7100-72df-11e9-919b-fc9b1862feb7.png)
